# ML-02 — Research Question and Provisional Lane

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

## My Lane: Content Performance Optimization

I chose the Content Performance Optimization lane because it focuses on understanding which page characteristics are associated with better search performance. My goal is to identify patterns that may help content teams decide which pages should be improved first. This lane is useful because it supports practical decision-making using measurable data instead of assumptions.

## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

## The Question: Decision, Action, Cost of a Wrong Call

The decision I want to improve is identifying which content pages should be prioritized for optimization.

The people who act on this information are SEO specialists and content teams.

If the recommendation is wrong, time and effort may be spent improving pages that have little impact, while pages with greater improvement potential may be ignored.

Machine learning can support this decision by identifying patterns in the data and providing decision-support based on observed relationships.

## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [11]:
import pandas as pd

# Load the FlyRank starter dataset
df = pd.read_csv("data/raw/content_refresh_anonymized.csv")

print("Number of pages:", len(df))
print("Number of columns:", len(df.columns))

print("\nFirst 10 column names:")
print(df.columns[:10].tolist())

Number of pages: 30000
Number of columns: 44

First 10 column names:
['content_id', 'client_id', 'search_volume', 'competition', 'competition_level', 'cpc', 'content_type', 'main_intent', 'word_count', 'char_count']


In [12]:
!git clone https://github.com/Aaronabraham1307/flyrank-ml-aaron.git
%cd flyrank-ml-aaron

Cloning into 'flyrank-ml-aaron'...
remote: Enumerating objects: 115, done.
remote: Counting objects: 100% (115/115), done.
remote: Compressing objects: 100% (86/86), done.
remote: Total 115 (delta 32), reused 77 (delta 13), pack-reused 0 (from 0)
Receiving objects: 100% (115/115), 1.84 MiB | 5.06 MiB/s, done.
Resolving deltas: 100% (32/32), done.
/content/flyrank-ml-aaron/flyrank-ml-aaron/flyrank-ml-aaron/flyrank-ml-aaron


In [13]:
print("Total pages:", len(df))
print("Average word count:", round(df["word_count"].mean(), 2))
print("Average search volume:", round(df["search_volume"].mean(), 2))

Total pages: 30000
Average word count: 3107.76
Average search volume: 158.88


## Quick Look at the Data

The starter dataset contains 30,000 pages with 44 columns.

The average page contains about 3,107.76 words, and the average search volume is 158.88. These numbers show that the dataset is large enough to explore patterns between page characteristics and search performance, making it suitable for this project.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

## Careful Words

This project will identify observed relationships between page characteristics and search performance. The results can support content teams in making better decisions about which pages to improve.

This project cannot prove that one factor causes better search performance, and it cannot predict Google's ranking algorithm. The findings are based only on the available dataset.

## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.